In [1]:
import pandas as pd
import plotly.express as px
import plotly.graph_objects as go



## Contexte
"swipe right" lorsque l'utilisateur aime le profil visionné
"swipe left" lorsque l'utilisateur n'aime pas le profil visionné

On cherche à comprendre via ce dataset ce qui rend les gens intéréssés les uns des autres
Ce dataset est basé sur une expérience de "speed dating"
Chaque ligne représente un "speed dating" et indique si chacunes des deux personnes accepterait un 2ème RDV


## Objectifs
comprendre ce qui intéresse les gens les uns par rapport aux autres pour aller au 2ème RDV

## Déroulé de l'experience
chaque participant a un 1er RDV de 4 minutes avec tous les autres participants du sexe opposé
A la fin des 4 minutes, on demande aux participants s'ils souhaitent aller au second RDV ET ils notent leur date sur 6 attributs (critères):
- attractivité = attr
- sincérité = sinc
- intelligence = intel
- fun = fun
- ambition = amb
- intérets communs = shar


NOTES D'ATTRACTIVITE:
- attr1_1: en quoi l'attirance physique est importante pour vous dans un RDV
- attr: à quel point vous avez trouvé attirant votre date
- attr_0: à quel point votre date vous a trouvé attirant

NOTES DE SINCERITE:
- sinc1_1: en quoi la sincérité est importante pour vous dans un RDV
- sinc: à quel point vous avez trouvé votre date sincère
- sinc_0: à quel point votre date vous a trouvé sincère

...

## Idées d'exploration
- Quels sont les attributs les moins souhaités chez un partenaire masculin 
- Quels sont les attributs les moins souhaités chez un partenaire féminin
- Sont-ils différents
- Les intérêts communs sont-ils plus importants que la race de chaque participants ?
- Pour obtenir un second RDV, est-il préférable de faire le 1er RDV en début de soirée ou en fin de soirée ?


### EXPLORATION DES DONNEES

In [2]:
df = pd.read_csv("Speed+Dating+Data.csv", encoding='Latin-1')
df.head()

iid   id  gender  idg  condtn  wave  round  position  positin1  order  ...  \
0    1  1.0       0    1       1     1     10         7       NaN      4  ...   
1    1  1.0       0    1       1     1     10         7       NaN      3  ...   
2    1  1.0       0    1       1     1     10         7       NaN     10  ...   
3    1  1.0       0    1       1     1     10         7       NaN      5  ...   
4    1  1.0       0    1       1     1     10         7       NaN      7  ...   

   attr3_3  sinc3_3  intel3_3  fun3_3  amb3_3  attr5_3  sinc5_3  intel5_3  \
0      5.0      7.0       7.0     7.0     7.0      NaN      NaN       NaN   
1      5.0      7.0       7.0     7.0     7.0      NaN      NaN       NaN   
2      5.0      7.0       7.0     7.0     7.0      NaN      NaN       NaN   
3      5.0      7.0       7.0     7.0     7.0      NaN      NaN       NaN   
4      5.0      7.0       7.0     7.0     7.0      NaN      NaN       NaN   

   fun5_3  amb5_3  
0     NaN     NaN  
1     NaN     NaN  
2     NaN     NaN  
3     NaN     NaN  
4     NaN     NaN  

[5 rows x 195 columns]

In [3]:
df.shape

(8378, 195)

In [4]:
pd.set_option('display.max_columns', 195)
df.head()

iid   id  gender  idg  condtn  wave  round  position  positin1  order  \
0    1  1.0       0    1       1     1     10         7       NaN      4   
1    1  1.0       0    1       1     1     10         7       NaN      3   
2    1  1.0       0    1       1     1     10         7       NaN     10   
3    1  1.0       0    1       1     1     10         7       NaN      5   
4    1  1.0       0    1       1     1     10         7       NaN      7   

   partner   pid  match  int_corr  samerace  age_o  race_o  pf_o_att  \
0        1  11.0      0      0.14         0   27.0     2.0      35.0   
1        2  12.0      0      0.54         0   22.0     2.0      60.0   
2        3  13.0      1      0.16         1   22.0     4.0      19.0   
3        4  14.0      1      0.61         0   23.0     2.0      30.0   
4        5  15.0      1      0.21         0   24.0     3.0      30.0   

   pf_o_sin  pf_o_int  pf_o_fun  pf_o_amb  pf_o_sha  dec_o  attr_o  sinc_o  \
0      20.0      20.0      20.0       0.0       5.0      0     6.0     8.0   
1       0.0       0.0      40.0       0.0       0.0      0     7.0     8.0   
2      18.0      19.0      18.0      14.0      12.0      1    10.0    10.0   
3       5.0      15.0      40.0       5.0       5.0      1     7.0     8.0   
4      10.0      20.0      10.0      10.0      20.0      1     8.0     7.0   

   intel_o  fun_o  amb_o  shar_o  like_o  prob_o  met_o   age field  field_cd  \
0      8.0    8.0    8.0     6.0     7.0     4.0    2.0  21.0   Law       1.0   
1     10.0    7.0    7.0     5.0     8.0     4.0    2.0  21.0   Law       1.0   
2     10.0   10.0   10.0    10.0    10.0    10.0    1.0  21.0   Law       1.0   
3      9.0    8.0    9.0     8.0     7.0     7.0    2.0  21.0   Law       1.0   
4      9.0    6.0    9.0     7.0     8.0     6.0    2.0  21.0   Law       1.0   

  undergra mn_sat tuition  race  imprace  imprelig     from zipcode  \
0      NaN    NaN     NaN   4.0      2.0       4.0  Chicago  60,521   
1      NaN    NaN     NaN   4.0      2.0       4.0  Chicago  60,521   
2      NaN    NaN     NaN   4.0      2.0       4.0  Chicago  60,521   
3      NaN    NaN     NaN   4.0      2.0       4.0  Chicago  60,521   
4      NaN    NaN     NaN   4.0      2.0       4.0  Chicago  60,521   

      income  goal  date  go_out  career  career_c  sports  tvsports  \
0  69,487.00   2.0   7.0     1.0  lawyer       NaN     9.0       2.0   
1  69,487.00   2.0   7.0     1.0  lawyer       NaN     9.0       2.0   
2  69,487.00   2.0   7.0     1.0  lawyer       NaN     9.0       2.0   
3  69,487.00   2.0   7.0     1.0  lawyer       NaN     9.0       2.0   
4  69,487.00   2.0   7.0     1.0  lawyer       NaN     9.0       2.0   

   exercise  dining  museums  art  hiking  gaming  clubbing  reading   tv  \
0       8.0     9.0      1.0  1.0     5.0     1.0       5.0      6.0  9.0   
1       8.0     9.0      1.0  1.0     5.0     1.0       5.0      6.0  9.0   
2       8.0     9.0      1.0  1.0     5.0     1.0       5.0      6.0  9.0   
3       8.0     9.0      1.0  1.0     5.0     1.0       5.0      6.0  9.0   
4       8.0     9.0      1.0  1.0     5.0     1.0       5.0      6.0  9.0   

   theater  movies  concerts  music  shopping  yoga  exphappy  expnum  \
0      1.0    10.0      10.0    9.0       8.0   1.0       3.0     2.0   
1      1.0    10.0      10.0    9.0       8.0   1.0       3.0     2.0   
2      1.0    10.0      10.0    9.0       8.0   1.0       3.0     2.0   
3      1.0    10.0      10.0    9.0       8.0   1.0       3.0     2.0   
4      1.0    10.0      10.0    9.0       8.0   1.0       3.0     2.0   

   attr1_1  sinc1_1  intel1_1  fun1_1  amb1_1  shar1_1  attr4_1  sinc4_1  \
0     15.0     20.0      20.0    15.0    15.0     15.0      NaN      NaN   
1     15.0     20.0      20.0    15.0    15.0     15.0      NaN      NaN   
2     15.0     20.0      20.0    15.0    15.0     15.0      NaN      NaN   
3     15.0     20.0      20.0    15.0    15.0     15.0      NaN      NaN   
4     15.0     

In [5]:
df.describe(include='all')

iid           id       gender          idg       condtn  \
count   8378.000000  8377.000000  8378.000000  8378.000000  8378.000000   
unique          NaN          NaN          NaN          NaN          NaN   
top             NaN          NaN          NaN          NaN          NaN   
freq            NaN          NaN          NaN          NaN          NaN   
mean     283.675937     8.960248     0.500597    17.327166     1.828837   
std      158.583367     5.491329     0.500029    10.940735     0.376673   
min        1.000000     1.000000     0.000000     1.000000     1.000000   
25%      154.000000     4.000000     0.000000     8.000000     2.000000   
50%      281.000000     8.000000     1.000000    16.000000     2.000000   
75%      407.000000    13.000000     1.000000    26.000000     2.000000   
max      552.000000    22.000000     1.000000    44.000000     2.000000   

               wave        round     position     positin1        order  \
count   8378.000000  8378.000000  8378.000000  6532.000000  8378.000000   
unique          NaN          NaN          NaN          NaN          NaN   
top             NaN          NaN          NaN          NaN          NaN   
freq            NaN          NaN          NaN          NaN          NaN   
mean      11.350919    16.872046     9.042731     9.295775     8.927668   
std        5.995903     4.358458     5.514939     5.650199     5.477009   
min        1.000000     5.000000     1.000000     1.000000     1.000000   
25%        7.000000    14.000000     4.000000     4.000000     4.000000   
50%       11.000000    18.000000     8.000000     9.000000     8.000000   
75%       15.000000    20.000000    13.000000    14.000000    13.000000   
max       21.000000    22.000000    22.000000    22.000000    22.000000   

            partner          pid        match     int_corr     samerace  \
count   8378.000000  8368.000000  8378.000000  8220.000000  8378.000000   
unique          NaN          NaN          NaN          NaN          NaN   
top             NaN          NaN          NaN          NaN          NaN   
freq            NaN          NaN          NaN          NaN          NaN   
mean       8.963595   283.863767     0.164717     0.196010     0.395799   
std        5.491068   158.584899     0.370947     0.303539     0.489051   
min        1.000000     1.000000     0.000000    -0.830000     0.000000   
25%        4.000000   154.000000     0.000000    -0.020000     0.000000   
50%        8.000000   281.000000     0.000000     0.210000     0.000000   
75%       13.000000   408.000000     0.000000     0.430000     1.000000   
max       22.000000   552.000000     1.000000     0.910000     1.000000   

              age_o       race_o     pf_o_att     pf_o_sin     pf_o_int  \
count   8274.000000  8305.000000  8289.000000  8289.000000  8289.000000   
unique          NaN          NaN          NaN          NaN          NaN   
top             NaN          NaN          NaN          NaN          NaN   
freq            NaN          NaN          NaN          NaN          NaN   
mean      26.364999     2.756653    22.495347    17.396867    20.270759   
std        3.563648     1.230689    12.569802     7.044003     6.782895   
min       18.000000     1.000000     0.000000     0.000000     0.000000   
25%       24.000000     2.000000    15.000000    15.000000    17.390000   
50%       26.000000     2.000000    20.000000    18.370000    20.000000   
75%       28.000000     4.000000    25.000000    20.000000    23.810000   
max       55.000000     6.000000   100.000000    60.000000    50.000000   

           pf_o_fun     pf_o_amb     pf_o_sha        dec_o       attr_o  \
count   8280.000000  8271.000000  8249.000000  8378.000000  8166.000000   
unique          NaN          NaN          NaN          NaN          NaN   
top             NaN          NaN          NaN          NaN          NaN   
freq            NaN          NaN          NaN          NaN          NaN   
mean      17.459714    10.685375    1

### On regarde les valeurs manquantes

In [6]:

df.isnull().sum()

iid            0
id             1
gender         0
idg            0
condtn         0
            ... 
attr5_3     6362
sinc5_3     6362
intel5_3    6362
fun5_3      6362
amb5_3      6362
Length: 195, dtype: int64

## VISUALISTION GLOBALE AIDANT A LA COMPREHENSION DES DONNEES 

### Distribution de l'age des participants

In [7]:
px.histogram(df, x='age', nbins=100)

### On regarde combien de personne ont trouvé un match lors de cette expérience

In [8]:
df['match'].value_counts() # il semblerait que 1380 personnes ont eu un second RDV et 6998 n'ont pas été au second RDV

match
0    6998
1    1380
Name: count, dtype: int64

In [9]:
percent_match = df['match'].sum() / df['match'].count() * 100
percent_match

16.47171162568632

16.5% des personnes ont eu un match, ce qui est peu

In [10]:
px.pie(df, names='match', title='Proportion de match')

### Ce que les utilisateurs de tinder cherchent en utilisant l'application

In [11]:

# Le but du speed dating pour les participants
display(df['goal'].value_counts().reset_index())
fig = px.histogram(df, x='goal',nbins=15, category_orders=dict({'goal':['Get a fun night', 'Meet new people', 'Get a date', 'To say "I did it"', 'Other', 'Looking for serious relationship']}))
fig.show()

goal  count
0   1.0   3426
1   2.0   3012
2   3.0    631
3   5.0    510
4   6.0    419
5   4.0    301

In [12]:
df["goal"] #type float, je veux des type str pour pouvoir changer les valeurs pour que ce soit plus explicite

0       2.0
1       2.0
2       2.0
3       2.0
4       2.0
       ... 
8373    1.0
8374    1.0
8375    1.0
8376    1.0
8377    1.0
Name: goal, Length: 8378, dtype: float64

In [13]:
df['goal'] = df['goal'].astype(str)

In [14]:
df['goal'] = df['goal'].replace({'1.0':'Get a fun night', '2.0':'Meet people', '3.0':'Get a date', '4.0':'Looking serious relationship', '5.0':'To say i did it', '6.0':'Other'})


In [15]:
df['goal']

0           Meet people
1           Meet people
2           Meet people
3           Meet people
4           Meet people
             ...       
8373    Get a fun night
8374    Get a fun night
8375    Get a fun night
8376    Get a fun night
8377    Get a fun night
Name: goal, Length: 8378, dtype: object

In [16]:
display(df['goal'].value_counts().reset_index())
fig = px.histogram(df, x='goal',nbins=15, title="What users are looking for", color='goal')

goal  count
0               Get a fun night   3426
1                   Meet people   3012
2                    Get a date    631
3               To say i did it    510
4                         Other    419
5  Looking serious relationship    301
6                           nan     79

In [17]:
fig.update_layout(showlegend=False)
fig.show()

### Ce que cherchent les femmes chez un homme

In [18]:
df.groupby('gender')['pf_o_att'].value_counts()

gender  pf_o_att
0       20.00       840
        25.00       557
        30.00       553
        40.00       285
        50.00       198
                   ... 
1       45.00        10
        2.00          9
        90.00         9
        6.67          5
        12.77         5
Name: count, Length: 114, dtype: int64

In [19]:
display(df[df['gender']==0]['pf_o_att'].mean())
display(df[df['gender']==1]['pf_o_att'].mean())

26.893883285302593

18.05522424242424